Src/WindowsServer/WindowsServer.Shared.Tests/AzureWebAppRoleEnvironmentTelemetryInitializerTest.cs

```diff
@@ -21,77 +21,98 @@ public void AzureWebAppRoleEnvironmentTelemetryInitializerSetsRoleName()
 
-            Environment.SetEnvironmentVariable("WEBSITE_HOSTNAME", null);
+            Environment.SetEnvironmentVariable(testVarName, null);
         }
 
         [TestMethod]
         public void AzureWebAppRoleEnvironmentTelemetryInitializerDoesNotOverrideRoleName()
         {
-            Environment.SetEnvironmentVariable("WEBSITE_HOSTNAME", "TestRoleName.azurewebsites.net");
+            var testVarName = "WEBSITE_" + Guid.NewGuid().ToString() + "_HOSTNAME";
+            Environment.SetEnvironmentVariable(testVarName, "TestRoleName.azurewebsites.net");
 
             var telemetryItem = new EventTelemetry();
             telemetryItem.Context.Cloud.RoleName = "Test";
 
-            var initializer = new AzureWebAppRoleEnvironmentTelemetryInitializer();
+            var initializer = new AzureWebAppRoleEnvironmentTelemetryInitializer()
+            {
+                WebAppHostNameEnvironmentVariable = testVarName
+            };
             initializer.Initialize(telemetryItem);
 
             Assert.Equal("Test", telemetryItem.Context.Cloud.RoleName);            
             Assert.Equal("TestRoleName.azurewebsites.net", telemetryItem.Context.GetInternalContext().NodeName);
 
-            Environment.SetEnvironmentVariable("WEBSITE_HOSTNAME", null);
+            Environment.SetEnvironmentVariable(testVarName, null);
         }
 
         [TestMethod]
         public void AzureWebAppRoleEnvironmentTelemetryInitializerDoesNotOverrideRoleInstance()
         {
-            Environment.SetEnvironmentVariable("WEBSITE_HOSTNAME", "TestRoleName.azurewebsites.net");
+            var testVarName = "WEBSITE_" + Guid.NewGuid().ToString() + "_HOSTNAME";
+            Environment.SetEnvironmentVariable(testVarName, "TestRoleName.azurewebsites.net");
 
             var telemetryItem = new EventTelemetry();
             telemetryItem.Context.Cloud.RoleInstance = "Test";
 
-            var initializer = new AzureWebAppRoleEnvironmentTelemetryInitializer();
+            var initializer = new AzureWebAppRoleEnvironmentTelemetryInitializer()
+            {
+                WebAppHostNameEnvironmentVariable = testVarName
+            };
             initializer.Initialize(telemetryItem);
 
             Assert.Equal("TestRoleName", telemetryItem.Context.Cloud.RoleName);
             Assert.Equal("Test", telemetryItem.Context.Cloud.RoleInstance);
             Assert.Equal("TestRoleName.azurewebsites.net", telemetryItem.Context.GetInternalContext().NodeName);
 
-            Environment.SetEnvironmentVariable("WEBSITE_HOSTNAME", null);
+            Environment.SetEnvironmentVariable(testVarName, null);
         }
 
         [TestMethod]
         public void AzureWebAppRoleEnvironmentTelemetryInitializerDoesNotOverrideNodeName()
         {
-            Environment.SetEnvironmentVariable("WEBSITE_HOSTNAME", "TestRoleName.azurewebsites.net");
-            
+            var testVarName = "WEBSITE_" + Guid.NewGuid().ToString() + "_HOSTNAME";
```

Comment:
Why do we change the ENV vars?

Reply:
We now have 3 classes that all use the *same environment variables* and testing each of them means we update/clear/change that small set of environment variables as we go. If we run the tests in parallel, tests will have that value changed unexpectedly during their execution by other tests.